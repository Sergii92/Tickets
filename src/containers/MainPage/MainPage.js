import React from 'react'
import styled from 'styled-components';

import Session from '../../components/Session/Session'

const MainPageWrapper = styled.div`
display:flex;
flex-wpar:wrap;
max-width:1200px;
background-color: black;
position:relative;
left:50%;
transform:translateX(-50%)
`


const MainPage = ()=>{
  return(
<MainPageWrapper>
  <Session/>
</MainPageWrapper>
  )
}

export default MainPage