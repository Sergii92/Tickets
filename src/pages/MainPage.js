import React, { useEffect } from "react";
import { useSelector, useDispatch } from "react-redux";
import { dataActions } from "../reducers/Session/actions";
import { selectData } from "../reducers/Session/selectors";

import Day from "../components/Day";
import { MainPageWrapper } from "../styled/pages";

// import { data } from "../constants/data";

const MainPage = () => {
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(dataActions.fetchData());
  }, []);

  const data = useSelector(selectData);

  return (
    <MainPageWrapper>
      {data && data.map((item) => <Day item={item} key={item.id} />)}
    </MainPageWrapper>
  );
};

export default MainPage;
