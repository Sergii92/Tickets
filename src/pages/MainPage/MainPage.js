import React, { useState, useEffect } from "react";
import styled from "styled-components";

import { data } from "../../constants/data";
import Day from "../../components/Day/Day";

const MainPageWrapper = styled.div`
  display: flex;
  flex-wrap: wrap;
  justify-content: space-around;
  width: 80%;
  max-width: 1400px;
  background-color: white;
  position: absolute;
  padding-top: 20px;
  padding-bottom: 20px;
  margin-top: 20px;
  margin-bottom: 20px;
  left: 50%;

  transform: translate(-50%);
`;

const MainPage = () => {
  const [sessionDays, setsessionDays] = useState(null);

  const getSessionDays = async () => {
    try {
      const responce = await fetch(" http://localhost:3004/sessionsDays");
      const data = await responce.json();
      setsessionDays(data);
    } catch (error) {
      console.log(error);
    }
  };
  useEffect(() => {
    getSessionDays();
  }, []);

  console.log(sessionDays);

  return (
    <MainPageWrapper>
      {data.sessionsDays.map((item) => {
        return <Day item={item} key={item.id}></Day>;
      })}
    </MainPageWrapper>
  );
};

export default MainPage;
