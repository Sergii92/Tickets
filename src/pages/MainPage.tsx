import React from "react";

import Day from "../components/Day";
import { MainPageWrapper } from "../styled/pages";
import { Iday } from "../interfaces";

interface Props {
  days: Array<Iday>;
}

const MainPage = ({ days }: Props) => {
  return (
    <MainPageWrapper>
      {days && days.map((day: Iday) => <Day day={day} key={day.id} />)}
    </MainPageWrapper>
  );
};

export default MainPage;
