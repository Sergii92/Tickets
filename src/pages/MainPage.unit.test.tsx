import React from "react";
import { shallow } from "enzyme";
import MainPage from "./MainPage";

describe("MainPage Component", () => {
  const daysData = [
    { id: 1, name: "dfdf" },
    { id: 2, name: "dfdf" },
    { id: 3, name: "dfdf" },
    { id: 4, name: "dfdf" },
  ];

  it("Renders", () => {
    const component = shallow(<MainPage days={daysData} />);
    expect(component).toMatchSnapshot();
  });

  it("Test data", () => {
    const data = jest.fn(() => true);
    data();
    expect(data).toHaveReturned();
    expect(daysData).toHaveLength(daysData.length);
  });
});
