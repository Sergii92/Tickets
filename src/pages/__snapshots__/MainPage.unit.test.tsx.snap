// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainPage Component Renders 1`] = `ShallowWrapper {}`;

exports[`MainPage Component should render MainPage without props 1`] = `ShallowWrapper {}`;
